### PR TITLE
featutre(web_services): adds a plugins.search API method

### PR DESCRIPTION
**Pull request by hypeJunction**
Web services can now be used to search for plugins

### CODE REVIEW

**Comment by mrclay** [comment]
Cool. What's the motivation?

**Comment by hypeJunction** [comment]
Working an installer plugin a-la WordPress.

**Comment by hypeJunction** [comment]
Still work in progress, but you can take it for a spin: https://github.com/hypeJunction/hypePluginInstaller

File: start.php [hunk 1]
```diff
@@ -107,21 +140,346 @@ function community_ws_plugin_check($plugins, $version) {
...
+		if (!$release) {
+			$release = $plugin->getRecentReleaseByElggVersion((string) $version);
+		}
+		if (!$release) {
+			continue;
+		}
+		$result['items'][] = $release->toObject();
+	}
+
+	return $result;
```
Comment (hypeJunction) [inline start.php:439]: Need to update results array to match the non-solr handler.

**Pull request closed without merging**